docs/lab/lab04/lab04.md

```diff
@@ -0,0 +1,384 @@
+
+:::info
+The RP2040 timer is fully monotomic, meaning it can never truly overflow. Its value is stored on 64 bits and increments every 1 microsecond, which means that the last value it can increment to before overflowing is 2^64-1, which the equivalent of roughly 500.000 years. This timer allows 4 different alarms, which can be used independently (TIMER_IRQ_0/1/2/3).
+:::
+
+## Analog and Digital Signals
+
+**Analog signals** are a representation of real-world data. They communicate information in a continuous function of time. They are smooth and time-varying waves, and contain an infinite number of values within the continuous range. An example of an analog signal would be sound, or the human voice.
+
+![AnalogSignal](images/analog_signal.png)
+
+**Digital signals** are a discrete representation of data. They are represented by a sequence of binary values, taken from a finite set of possible numbers. They are square and discrete waves. In most cases, they are represented by two values: 0 and 1 (or 0V and 5V). Digital representation of signals is usually used in hardware.
+
+![DigitalSignal](images/digital_signal.png)
+
+## Pulse-Width Modulation (PWM)
+Up to now, we learned to turn a led on and off, or in other words, set a led's intensity to 100% or 0%. What if we wanted to turn on the led only at 50% intensity? We only have a two-level digital value, 0 or 1, so technically a value of 0.5 is not possible. What we can do is simulate this analog signal, so that it *looks* like the led is at half intensity. 
+
+**Pulse-Width Modulation** is a method of simulating an analog signal using a digital one, by varying the width of the generated square wave. 
+
+![PWMExample](images/pulse-width-modulation-signal-diagrams-average.png)
+
+:::note
+We can think of the simulated analog signal being directly proportional to the change in digital signal pulse size. The larger the square wave at a given period T, the higher the average analog amplitude output for that period.
+:::
+
+The **duty cycle** of the signal is the percentage of time per period that the signal is high.
+
+![DutyCycle](images/duty_cycle.png)
+
+So if we wanted our led to be at 50% intensity, we would choose a duty cycle of 50%. By quickly switching between high and low, the led appears to the human eye as being at only 50% intensity, when in reality, it's only on at max intensity 50% of the time, and off the rest of the time.
+
+![PWMLed](images/pwm_led.gif)
+
+$$
+
+duty\_cycle = \frac{time\_on}{period} \%
+
+$$
+
+For the RP2040, to generate this PWM signal, a *counter* is used. When the counter is reset, the value of the output signal is 1. The counter counts up until it reaches a certain value, after which the value of the output signal becomes 0. The counter continues to count until the top (or until it overflows), and then the signal becomes 1 again. This way, by choosing the value that the counter should compare to, we set the duty cycle of the PWM signal.
```

Comment:
I would use the actual register names for the RP2040.

Reply:
Done, it's clearer and easier to follow now, I think.